src/vpk.cpp

```diff
@@ -95,6 +122,34 @@ namespace VpkParser {
     return std::move(fileData);
   }
 
+  std::optional<std::pair<std::vector<std::filesystem::path>, std::vector<std::filesystem::path>>> Vpk::list(
+    const std::filesystem::path& path
+  ) const {
+    std::vector<std::filesystem::path> fileList = {};
+    std::vector<std::filesystem::path> directoryList = {};
+
+    for (const auto& [extension, directories] : files) {
+      for (const auto& [dir, fileNames] : directories) {
+        auto dirPath = normalizePath(std::filesystem::path(dir));
+
+        if (isPathSubfolderOfBase(dirPath, path)) {
+          directoryList.push_back(dirPath.filename());
+        } else if (dirPath == path) {
+          for (const auto& fileName : fileNames | std::views::keys) {
+            fileList.emplace_back(fileName + extension);
+          }
+        }
+      }
+    }
+
+    if (fileList.empty() && directoryList.empty()) {
+      return std::nullopt;
```

Comment:
Null should mean "the path you gave me doesnt exist"

This means "the path you gave me either doesnt exist or exists but is empty"

The empty case should return both vectors left empty